#**Curso: Lógica de Programação e Algoritmos com Python e exercícios!**
link: https://www.udemy.com/share/102b2oBUEacl1bTHw=/


Neste curso foi abordado alguns exercícios provenientes da wiki do Python Brasil. Neste notebook ficarão registrados a minha resolução para os exercícios abordados no curso.

link dos exercícios: https://wiki.python.org.br/ListaDeExercicios

#**ESTRUTURA SEQUENCIAL**

##Tarefa 01 - Faça um Programa que mostre a mensagem "Alo mundo" na tela.

In [ ]:
print("Alo mundo")

Alo mundo


##Tarefa 02 - Faça um Programa que peça um número e então mostre a mensagem "O número informado foi [número]".

In [ ]:
num = float(input('Digite um numero: '))
print(f'O numero informado foi {num:.0f}')

Digite um numero: 15
O numero informado foi 15


##Tarefa 03 - Faça um Programa que peça dois números e imprima a soma.

In [ ]:
a = float(input("Digite um número: "))
b = float(input("Digite outro número: "))
soma = a + b
print(f'A soma dos números é {soma}')

Digite um número: 5
Digite outro número: 8
A soma dos números é 13.0


##Tarefa 04 - Faça um Programa que peça as 4 notas bimestrais e mostre a média.

In [ ]:
bim1 = float(input("Digite a nota do 1º Bimestre: "))
bim2 = float(input("Digite a nota do 2º Bimestre: "))
bim3 = float(input("Digite a nota do 3º Bimestre: "))
bim4 = float(input("Digite a nota do 4º Bimestre: "))
mediaNotas = (bim1+bim2+bim3+bim4)/4
print(f'A médias das notas bimestrais é {mediaNotas}')

Digite a nota do 1º Bimestre: 5
Digite a nota do 2º Bimestre: 9
Digite a nota do 3º Bimestre: 4
Digite a nota do 4º Bimestre: 7
A médias das notas bimestrais é 6.25


##Tarefa 05 - Faça um Programa que converta metros para centímetros.

In [ ]:
metros = float(input("Digite a distância em metros: "))
centimetros = metros*100
print(f'A distância informada em centímetros é {centimetros:.0f}')

Digite a distância em metros: 20
A distância informada em centímetros é 2000


##Tarefa 06 - Faça um Programa que peça o raio de um círculo, calcule e mostre sua área.

In [ ]:
import math

raio = float(input("Digite a a distância do raio em metros: "))
area = math.pi*(raio**2)
print(f'A área da circunferência é {area:.2f}m²')

Digite a a distância do raio em metros: 2
A área da circunferência é 12.57m²


##Tarefa 07 - Faça um Programa que calcule a área de um quadrado, em seguida mostre o dobro desta área para o usuário.

In [ ]:
lado = float(input("Digite o valor em metros de um dos lados do quadrado: "))
area = lado * lado
print(f'O dobro da área deste quadrado é {area*2:.2f}m²')

Digite o valor em metros de um dos lados do quadrado: 5
O dobro da área deste quadrado é 50.00m²


##Tarefa 08 - Faça um Programa que pergunte quanto você ganha por hora e o número de horas trabalhadas no mês. Calcule e mostre o total do seu salário no referido mês.

In [ ]:
salarioHora = float(input("Digite quanto você recebe de salário por hora: "))
horas = float(input("Digite quantas horas você trabalha por mês: "))
salarioMensal = salarioHora * horas
print(f'O seu salário mensal é {salarioMensal:.2f}')

Digite quanto você recebe de salário por hora: 50
Digite quantas horas você trabalha por mês: 100
O seu salário mensal é 5000.00


##Tarefa 09 - Faça um Programa que peça a temperatura em graus Farenheit, transforme e mostre a temperatura em graus Celsius.

In [ ]:
farenheit = float(input("Digite o valor da temperatura em graus Farenheit: "))
celsius = (5*(farenheit-32))/9
print(f'Esta temperatura em graus Celsius equivale a {celsius:.2f}')

Digite o valor da temperatura em graus Farenheit: 80
Esta temperatura em graus Celsius equivale a 26.67


##Tarefa 10 - Faça um Programa que peça a temperatura em graus Celsius, transforme e mostre em graus Farenheit. 

In [ ]:
celsius = float(input("Digite o valor da temperatura em graus Celsius: "))
farenheit = celsius*1.8 + 32
print(f'Esta temperatura em graus Farenheit equivale a {farenheit:.2f}')

Digite o valor da temperatura em graus Celsius: 32
Esta temperatura em graus Farenheit equivale a 89.60


##Tarefa 11 - Faça um Programa que peça 2 números inteiros e um número real. Calcule e mostre:
1.   o produto do dobro do primeiro com metade do segundo.
1.   a soma do triplo do primeiro com o terceiro.
2.   o terceiro elevado ao cubo.


In [ ]:
inte1 = int(input("Digite um número inteiro: "))
inte2 = int(input("Digite outro número inteiro: "))
real = float(input("Digite um número real: "))

calc1 = (2*inte1)*(inte2/2) 
calc2 = (3*inte1)+real
calc3 = real**3

print(f'o produto do dobro do primeiro com metade do segundo: {calc1:.2f}')
print(f'a soma do triplo do primeiro com o terceiro: {calc2:.2f}')
print(f'o terceiro elevado ao cubo: {calc3:.2f}')

Digite um número inteiro: 5
Digite outro número inteiro: 6
Digite um número real: 5.9
o produto do dobro do primeiro com metade do segundo: 30.00
a soma do triplo do primeiro com o terceiro: 20.90
o terceiro elevado ao cubo: 205.38


##Tarefa 12 - Tendo como dados de entrada a altura de uma pessoa, construa um algoritmo que calcule seu peso ideal, usando a seguinte fórmula: (72.7*altura) - 58.

In [ ]:
altura = float(input("Digite a sua altura: "))
pesoIdeal = (72.7*altura) - 58
print(f'Seu peso ideal: {pesoIdeal:.2f}')

Digite a sua altura: 1.74
Seu peso ideal: 68.50


##Tarefa 13 - Tendo como dado de entrada a altura (h) de uma pessoa, construa um algoritmo que calcule seu peso ideal, utilizando as seguintes fórmulas:
*   Para homens: (72.7*h) - 58.
*   Para mulheres: (62.1*h) - 44.7.


In [ ]:
altura = float(input("Digite a sua altura: "))
pesoIdealHomem = (72.7*altura) - 58
pesoIdealMulher = (62.1*altura) - 44.7
print(f'Seu peso ideal caso você seja homem: {pesoIdealHomem:.2f}')
print(f'Seu peso ideal caso você seja mulher: {pesoIdealMulher:.2f}')

Digite a sua altura: 1.74
Seu peso ideal caso você seja homem: 68.50
Seu peso ideal caso você seja mulher: 63.35


##Tarefa 14 - João Papo-de-Pescador, homem de bem, comprou um microcomputador para controlar o rendimento diário de seu trabalho. Toda vez que ele traz um peso de peixes maior que o estabelecido pelo regulamento de pesca do estado de São Paulo (50 quilos) deve pagar uma multa de R$ 4,00 por quilo excedente. João precisa que você faça um programa que leia a variável peso (peso de peixes) e calcule o excesso. Gravar na variável excesso a quantidade de quilos além do limite e na variável multa o valor da multa que João deverá pagar. Imprima os dados do programa com as mensagens adequadas.

In [ ]:
pesoPeixes = float(input("Digite o peso dos peixes em quilogramas: "))
excessoPeso = 0

if pesoPeixes > 50:
  excessoPeso = pesoPeixes-50

multa = excessoPeso*4
print(f'O peso que excedeu o limite é {excessoPeso:.2f}kg e deve-se pagar R${multa:.2f} em multa')

Digite o peso dos peixes em quilogramas: 53
O peso que excedeu o limite é 3.00kg e deve-se pagar R$12.00 em multa


##Tarefa 15 - Faça um Programa que pergunte quanto você ganha por hora e o número de horas trabalhadas no mês. Calcule e mostre o total do seu salário no referido mês, sabendo-se que são descontados 11% para o Imposto de Renda, 8% para o INSS e 5% para o sindicato, faça um programa que nos dê:
1.   salário bruto.
2.   quanto pagou ao INSS.
1.   quanto pagou ao sindicato.
2.   o salário líquido.
1.   calcule os descontos e o salário líquido, conforme a tabela abaixo:
*   Salário Bruto
*   IR (11%)
*   INSS (8%)
*   Sindicato ( 5%)
*   Salário Liquido (Salário Bruto - Descontos = Salário Líquido)


In [ ]:
salarioHora = float(input("Digite quanto você recebe de salário por hora: "))
horas = float(input("Digite quantas horas você trabalha por mês: "))
salarioMensalBruto = salarioHora * horas
ir = 0.11*salarioMensalBruto
inss = 0.08*salarioMensalBruto
sindicato = 0.05*salarioMensalBruto
salarioMensalLiquido = salarioMensalBruto - ir - inss - sindicato

print(f'+ Salário Bruto : R${salarioMensalBruto:.2f} \n'  
f'- IR (11%) : R${ir:.2f} \n'
f'- INSS (8%) : R${inss:.2f} \n'
f'- Sindicato (5%) : R${sindicato:.2f} \n'
f'= Salário Liquido : R${salarioMensalLiquido:.2f}')

Digite quanto você recebe de salário por hora: 50
Digite quantas horas você trabalha por mês: 150
+ Salário Bruto : R$7500.00 
- IR (11%) : R$825.00 
- INSS (8%) : R$600.00 
- Sindicato (5%) : R$375.00 
= Salário Liquido : R$5700.00


##Tarefa 16 - Faça um programa para uma loja de tintas. O programa deverá pedir o tamanho em metros quadrados da área a ser pintada. Considere que a cobertura da tinta é de 1 litro para cada 3 metros quadrados e que a tinta é vendida em latas de 18 litros, que custam R$ 80,00. Informe ao usuário a quantidades de latas de tinta a serem compradas e o preço total.

In [ ]:
area = float(input("Digite a área ser pintada em m²: "))
litroPorMetro = area/3
quantidadeLata = round((litroPorMetro/18)+0.5)
custoFinal = quantidadeLata * 80
print(f'A quantidade de lata(s) a ser(em) comprada(s) é {quantidadeLata} e o custo total é de R${custoFinal:.2f}.')

Digite a área ser pintada em m²: 120
A quantidade de lata(s) a ser(em) comprada(s) é 3 e o custo total é de R$240.00.


##Tarefa 17 - Faça um Programa para uma loja de tintas. O programa deverá pedir o tamanho em metros quadrados da área a ser pintada. Considere que a cobertura da tinta é de 1 litro para cada 6 metros quadrados e que a tinta é vendida em latas de 18 litros, que custam 80,00 reais ou em galões de 3,6 litros, que custam 25,00 reais.
1.  Informe ao usuário as quantidades de tinta a serem compradas e os respectivos preços em 3 situações:
*   comprar apenas latas de 18 litros;
*   comprar apenas galões de 3,6 litros;
*   misturar latas e galões, de forma que o preço seja o menor. Acrescente 10% de folga e sempre arredonde os valores para cima, isto é, considere latas cheias.


In [ ]:
area = float(input("Digite a área ser pintada em m²: "))
litroPorMetro = (area/6)*1.1

apenasLatas = round((litroPorMetro/18)+0.5)
custoApenasLatas = apenasLatas * 80

apenasGaloes = round((litroPorMetro/3.6)+0.5)
custoApenasGaloes = apenasGaloes * 25

#Latas e Galões
latas = apenasLatas
custoLatasGaloes = custoApenasLatas
if (litroPorMetro//18) <= 10.8:
  latas = int(litroPorMetro/18)
  restoLitros = litroPorMetro//18
  galoes = round((restoLitros/3.6)+0.5)
  custoLatasGaloes = (latas*80)+(galoes*25)

print(f'A quantidade de lata(s) a ser(em) comprada(s) é {apenasLatas} e o custo total é de R${custoApenasLatas:.2f}.')
print(f'A quantidade de galão(ões) a ser(em) comprado(s) é {apenasGaloes} e o custo total é de R${custoApenasGaloes:.2f}.')
print(f'Buscando o melhor preço, deve-se comprar {latas} lata(s) e {galoes} galão(ões), ao custo total de R${custoLatasGaloes:.2f}.')

Digite a área ser pintada em m²: 368
A quantidade de lata(s) a ser(em) comprada(s) é 4 e o custo total é de R$320.00.
A quantidade de galão(ões) a ser(em) comprado(s) é 19 e o custo total é de R$475.00.
Buscando o melhor preço, deve-se comprar 3 lata(s) e 1 galão(ões), ao custo total de R$265.00.


##Tarefa 18 - Faça um programa que peça o tamanho de um arquivo para download (em MB) e a velocidade de um link de Internet (em Mbps), calcule e informe o tempo aproximado de download do arquivo usando este link (em minutos).

In [ ]:
import math

tamanhoArquivo = float(input("Digite o tamanho do arquivo em MB: "))
velocidadeDownload = float(input("Digite a velocidade de download da sua internet em mbps: "))
tempoMinutos = (tamanhoArquivo/velocidadeDownload)/60
tempofinalmin = math.floor(tempoMinutos)
tempofinalseg = ((tempoMinutos%60) - tempofinalmin)*60
print(f'O tempo total para o download é de {tempofinalmin:.0f} minuto(s) e {tempofinalseg:.0f} segundo(s).')

Digite o tamanho do arquivo em MB: 1024
Digite a velocidade de download da sua internet em mbps: 1
O tempo total para o download é de 17 minuto(s) e 4 segundo(s).


#**ESTRUTURA DE DECISÃO**

##Tarefa 01 - Faça um Programa que peça dois números e imprima o maior deles.

In [ ]:
num1 = float(input("Digite um número: "))
num2 = float(input("Digite outro número: "))
maiorNumero = num1

if num2 > num1:
  maiorNumero = num2

print(f'O maior número digitado é {maiorNumero}')

Digite um número: 2
Digite outro número: 32
O maior número digitado é 32.0


##Tarefa 02 - Faça um Programa que peça um valor e mostre na tela se o valor é positivo ou negativo.

In [ ]:
valor = float(input('Digite um número: '))
resposta = "positivo"

if valor < 0:
  resposta = "negativo"

print(f"O valor digitado é {resposta}")

Digite um número: -36
O valor digitado é negativo


##Tarefa 03 - Faça um Programa que verifique se uma letra digitada é "F" ou "M". Conforme a letra escrever: F - Feminino, M - Masculino, Sexo Inválido.

In [ ]:
letraSexo = input("Digite a inicial do seu sexo: ")

if letraSexo.upper() == 'H':
  print("Homem")
elif letraSexo.upper() == 'M':
  print("Mulher")
else:
  print("Sexo inválido")

Digite a inicial do seu sexo: h
Homem


##Tarefa 04 - Faça um Programa que verifique se uma letra digitada é vogal ou consoante. 

In [ ]:
letra = input("Digite uma letra: ")

if letra.upper() == 'A' or letra.upper() == 'E' or letra.upper() == 'I' or letra.upper() == 'O' or letra.upper() == 'U':
  print("vogal")
else: 
  print("consoante")

Digite uma letra: y
consoante


##Tarefa 05 - Faça um programa para a leitura de duas notas parciais de um aluno. O programa deve calcular a média alcançada por aluno e apresentar:
*   A mensagem "Aprovado", se a média alcançada for maior ou igual a sete;
*   A mensagem "Reprovado", se a média for menor do que sete;
*   A mensagem "Aprovado com Distinção", se a média for igual a dez.

In [ ]:
nota1 = float(input("Digite a primeira nota: "))
nota2 = float(input("Digite a segunda nota: "))
media = (nota1+nota2)/2

if media == 10:
  print("Aprovado com Distinção")
elif media >= 7:
  print("Aprovado")
else:
  print("Reprovado")

Digite a primeira nota: 5
Digite a segunda nota: 4.5
Reprovado


##Tarefa 06 - Faça um Programa que leia três números e mostre o maior deles.

In [ ]:
num1 = float(input("Digite o primeiro número: "))
num2 = float(input("Digite o segundo número: "))
num3 = float(input("Digite o terceiro número: "))

if (num1 > num2) and (num1 > num3):
  print(num1)
elif (num2 > num1) and (num2 > num3):
  print(num2)
else:
  print(num3)

Digite o primeiro número: 5
Digite o segundo número: 9
Digite o terceiro número: 4
9.0


##Tarefa 07 - Faça um Programa que leia três números e mostre o maior e o menor deles.

In [ ]:
num1 = float(input("Digite o primeiro número: "))
num2 = float(input("Digite o segundo número: "))
num3 = float(input("Digite o terceiro número: "))

if (num1 > num2) and (num1 > num3):
  maior = num1
elif (num2 > num1) and (num2 > num3):
  maior = num2
else:
  maior = num3

if (num1 < num2) and (num1 < num3):
  menor = num1
elif (num2 < num1) and (num2 < num3):
  menor = num2
else:
  menor = num3

print(f'O maior número é {maior} e o menor númeoro é {menor}')

Digite o primeiro número: 8
Digite o segundo número: 2
Digite o terceiro número: 45
O maior número é 45.0 e o menor númeoro é 2.0


##Tarefa 08 - Faça um programa que pergunte o preço de três produtos e informe qual produto você deve comprar, sabendo que a decisão é sempre pelo mais barato.

In [ ]:
num1 = float(input("Digite o preço de primeiro produto: "))
num2 = float(input("Digite o preço de segundo produto: "))
num3 = float(input("Digite o preço de terceiro produto: "))

if (num1 < num2) and (num1 < num3):
  menor = num1
elif (num2 < num1) and (num2 < num3):
  menor = num2
else:
  menor = num3

print(f'Deve-se comprar o produto de valor R${menor:.2f}')

Digite o preço de primeiro produto: 56
Digite o preço de segundo produto: 24
Digite o preço de terceiro produto: 89
Deve-se comprar o produto de valor R$24.00


##Tarefa 09 - Faça um Programa que leia três números e mostre-os em ordem decrescente.

In [ ]:
num1 = float(input("Digite o primeiro número: "))
num2 = float(input("Digite o segundo número: "))
num3 = float(input("Digite o terceiro número: "))

lista = [num1,num2,num3]
lista.sort()
print(f'O maior número é {lista[2]}, o segundo maior é {lista[1]} e o menor é {lista[0]}')

Digite o primeiro número: 5
Digite o segundo número: 89
Digite o terceiro número: 12
O maior número é 89.0, o segundo maior é 12.0 e o menor é 5.0


##Tarefa 10 - Faça um Programa que pergunte em que turno você estuda. Peça para digitar M-matutino ou V-Vespertino ou N- Noturno. Imprima a mensagem "Bom Dia!", "Boa Tarde!" ou "Boa Noite!" ou "Valor Inválido!", conforme o caso.

In [ ]:
turno = input("Em qual turno você estuda? Responda com M-matutino, V-Vespertino ou N-Noturno ")

if turno.upper() == 'M':
  print('Bom Dia!')
elif turno.upper() == 'V':
  print("Boa Tarde!")
elif turno.upper() == 'N':
  print("Boa Noite!")
else:
  print("Valor Invávido!")

Em qual turno você estuda? Responda com M-matutino, V-Vespertino ou N-Noturno n
Boa Noite!


##Tarefa 11 - As Organizações Tabajara resolveram dar um aumento de salário aos seus colaboradores e lhe contrataram para desenvolver o programa que calculará os reajustes.

### Faça um programa que recebe o salário de um colaborador e o reajuste segundo o seguinte critério, baseado no salário atual:
*   salários até 280,00 reais (incluindo) : aumento de 20%
*   salários entre 280,00 e 700,00 reais  : aumento de 15%
*   salários entre 700,00 e 1500,00 reais  : aumento de 10%
*   salários de 1500,00 reais  em diante : aumento de 5% 
###Após o aumento ser realizado, informe na tela:
*   o salário antes do reajuste;
*   o percentual de aumento aplicado;
*   o valor do aumento;
*   o novo salário, após o aumento.


In [ ]:
salarioBase = float(input("Digite o valor do salário do funcionário: "))

if salarioBase <= 280:
  aumento = 20
elif salarioBase <= 700:
  aumento = 15
elif salarioBase <= 1500:
  aumento = 10
else:
  aumento = 5   

salarioNovo = salarioBase * ((aumento/100)+1)
valorAumento = salarioNovo - salarioBase

print(f'O salário antes do reajuste: R$ {salarioBase:.2f} \n'
f'O percentual de aumento aplicado: {aumento:.0f}% \n'
f'O valor do aumento: R$ {valorAumento:.2f} \n'
f'O salário após o aumento: R$ {salarioNovo:.2f}')

Digite o valor do salário do funcionário: 6000
O salário antes do reajuste: R$ 6000.00 
O percentual de aumento aplicado: 5% 
O valor do aumento: R$ 300.00 
O salário após o aumento: R$ 6300.00


##Tarefa 12 - Faça um programa para o cálculo de uma folha de pagamento, sabendo que os descontos são do Imposto de Renda, que depende do salário bruto (conforme tabela abaixo) e 10% para o INSS e que o FGTS corresponde a 11% do Salário Bruto, mas não é descontado (é a empresa que deposita). O Salário Líquido corresponde ao Salário Bruto menos os descontos. O programa deverá pedir ao usuário o valor da sua hora e a quantidade de horas trabalhadas no mês.

Desconto do IR:

Salário Bruto até 900 (inclusive) - isento

Salário Bruto até 1500 (inclusive) - desconto de 5%

Salário Bruto até 2500 (inclusive) - desconto de 10%

Salário Bruto acima de 2500 - desconto de 20% 

### Imprima na tela as informações, dispostas conforme o exemplo abaixo. No exemplo o valor da hora é 5 e a quantidade de hora é 220.
        Salário Bruto: (5 * 220)        : R$ 1100,00
        (-) IR (5%)                     : R$   55,00  
        (-) INSS ( 10%)                 : R$  110,00
        FGTS (11%)                      : R$  121,00
        Total de descontos              : R$  165,00
        Salário Liquido                 : R$  935,00

In [ ]:
salarioHora = float(input("Digite quanto você recebe de salário por hora: "))
horas = float(input("Digite quantas horas você trabalha por mês: "))
salarioMensalBruto = salarioHora * horas

if salarioMensalBruto <= 900:
  descontoIR = 0
elif salarioMensalBruto <= 1500:
  descontoIR = 5
elif salarioMensalBruto <= 2500:
  descontoIR = 10
else:
  descontoIR = 20   

ir = salarioMensalBruto * (descontoIR/100)
inss = 0.1*salarioMensalBruto
fgts = 0.11*salarioMensalBruto
salarioMensalLiquido = salarioMensalBruto - ir - inss

print(f'Salário Bruto: ({horas:.0f} * {salarioHora:.0f}) \t: R${salarioMensalBruto:.2f} \n'  
f'(-) IR ({descontoIR:.0f}%) \t\t\t: R${ir:.2f} \n'
f'(-) INSS (10%) \t\t\t: R${inss:.2f} \n'
f'FGTS (11%) \t\t\t: R${fgts:.2f} \n'
f'Salário Liquido \t\t: R${salarioMensalLiquido:.2f}')

Digite quanto você recebe de salário por hora: 50
Digite quantas horas você trabalha por mês: 150
Salário Bruto: (150 * 50) 	: R$7500.00 
(-) IR (20%) 			: R$1500.00 
(-) INSS (10%) 			: R$750.00 
FGTS (11%) 			: R$825.00 
Salário Liquido 		: R$5250.00


##Tarefa 13 - Faça um Programa que leia um número e exiba o dia correspondente da semana. (1-Domingo, 2- Segunda, etc.), se digitar outro valor deve aparecer valor inválido.

In [ ]:
numSemana = int(input("Digite o número da semana: "))
listaNomeSemana = ['Domingo', 'Segunda', 'Terça', 'Quarta', 'Quinta', 'Sexta', 'Sábado']

if numSemana in range(1, 8):
  print(listaNomeSemana[numSemana-1])
else:
  print("Valor Inválido")

Digite o número da semana: 7
Sábado


##Tarefa 14 - Faça um programa que lê as duas notas parciais obtidas por um aluno numa disciplina ao longo de um semestre, e calcule a sua média. A atribuição de conceitos obedece à tabela abaixo:

    Média de Aproveitamento  Conceito
    Entre 9.0 e 10.0         A
    Entre 7.5 e 9.0          B
    Entre 6.0 e 7.5          C
    Entre 4.0 e 6.0          D
    Entre 4.0 e zero         E

O algoritmo deve mostrar na tela as notas, a média, o conceito correspondente e a mensagem “APROVADO” se o conceito for A, B ou C ou “REPROVADO” se o conceito for D ou E.

In [ ]:
nota1 = float(input("Digite a primeira nota: "))
nota2 = float(input("Digite a segunda nota: "))
media = (nota1+nota2)/2

if media >= 9:
  conceito = "A"
  aprovacao = "APROVADO"
elif media >= 7.5:
  conceito = "B"
  aprovacao = "APROVADO"
elif media >= 6:
  conceito = "C"
  aprovacao = "APROVADO"
elif media >= 4:
  conceito = "D"
  aprovacao = "REPROVADO"
else:
  conceito = "E"
  aprovacao = "REPROVADO"

print(f'A média do aluno foi {media:.2f}, com o conceito {conceito}, assim estando {aprovacao} na disciplina.')

Digite a primeira nota: 5
Digite a segunda nota: 4.5
A média do aluno foi 4.75, com o conceito D, assim estando REPROVADO na disciplina.


##Tarefa 15 - Faça um Programa que peça os 3 lados de um triângulo. O programa deverá informar se os valores podem ser um triângulo. Indique, caso os lados formem um triângulo, se o mesmo é: equilátero, isósceles ou escaleno.

Dicas:
*   Três lados formam um triângulo quando a soma de quaisquer dois lados for maior que o terceiro;
*   Triângulo Equilátero: três lados iguais;
*   Triângulo Isósceles: quaisquer dois lados iguais;
*   Triângulo Escaleno: três lados diferentes.


In [ ]:
lado1 = float(input("Digite o valor do 1º lado do triângulo em centímetros: "))
lado2 = float(input("Digite o valor do 2º lado do triângulo em centímetros: "))
lado3 = float(input("Digite o valor do 3º lado do triângulo em centímetros: "))

if (lado1 + lado2 <= lado3) or (lado1 + lado3 <= lado2) or (lado3 + lado2 <= lado1):
  print("Os valores digitados não formam um triângulo")
else:
  if (lado1 == lado2 == lado3):
    formato = "Triângulo Equilátero"
  elif (lado1 == lado2) or (lado1 == lado3) or (lado3 == lado2):
    formato = "Triângulo Isósceles"
  else:
    formato = "Triângulo Escaleno"
  print(f'Os valores digitados formam um {formato}')


Digite o valor do 1º lado do triângulo em centímetros: 5
Digite o valor do 2º lado do triângulo em centímetros: 8
Digite o valor do 3º lado do triângulo em centímetros: 3
Os valores digitados não formam um triângulo


##Tarefa 16 - Faça um programa que calcule as raízes de uma equação do segundo grau, na forma ax² + bx + c. O programa deverá pedir os valores de a, b e c e fazer as consistências, informando ao usuário nas seguintes situações:
*   Se o usuário informar o valor de A igual a zero, a equação não é do segundo grau e o programa não deve fazer pedir os demais valores, sendo encerrado;
*   Se o delta calculado for negativo, a equação não possui raízes reais. Informe ao usuário e encerre o programa;
*   Se o delta calculado for igual a zero a equação possui apenas uma raiz real; informe-a ao usuário;
*   Se o delta for positivo, a equação possui duas raiz reais; informe-as ao usuário. 

In [ ]:
a = float(input("Para a equação ax² + bx + c = 0 digite o valor de a: "))

if a == 0:
  print("Quando a é igual a 0 a equação não é do segundo grau.")
else:
  b = float(input("Digite o valor de b: "))
  c = float(input("Digite o valor de c: "))
  if ((b**2)-(4*a*c)) < 0:
    print("A equação não possui raízes reais.")
  elif ((b**2)-(4*a*c)) == 0:
    x = (-b)/(2*a)
    print(f"A equação possui apenas uma raiz: x = {x:.2f}")
  else:
    x1 = (-b+(((b**2)-(4*a*c))**0.5))/(2*a)
    x2 = (-b-(((b**2)-(4*a*c))**0.5))/(2*a)
    print(f"A equação possui duas raizes: x1 = {x1:.2f} e x2 = {x2:.2f}")

Para a equação ax² + bx + c = 0 digite o valor de a: 1
Digite o valor de b: 14
Digite o valor de c: 2
A equação possui duas raizes: x1 = -0.14 e x2 = -13.86


##Tarefa 17 - Faça um Programa que peça um número correspondente a um determinado ano e em seguida informe se este ano é ou não bissexto.

In [ ]:
ano = float(input("Digite um ano: "))

if (ano%4 != 0):
  print("O ano digitado não é bissexto")
elif (ano%100 == 0) and (ano%400 != 0):
  print("O ano digitado não é bissexto")
else:
  print("O ano digitado é bissexto")

Digite um ano: 2022
O ano digitado não é bissexto


##Tarefa 18 - Faça um Programa que peça uma data no formato dd/mm/aaaa e determine se a mesma é uma data válida.

In [ ]:
dataDigitada = str(input("Digite uma data no formato dd/mm/aaaa: "))
dataDigitada = dataDigitada.split('/')
dia = int(dataDigitada[0])
mes = int(dataDigitada[1])
ano = int(dataDigitada[2])

def Bissexto(ano):
  if (ano%4 != 0):
    return False
  elif (ano%100 == 0) and (ano%400 != 0):
    return False
  else:
    return True

def ValidaAno(ano):
  if ano >= 1:
    return True
  else:
    return False

def ValidaMes(mes):
  if (mes >= 1) and (mes <= 12):
    return True
  else:
    return False

def ValidaDia(dia, mes):
  if mes in [1,3,5,7,8,10,12]:
    if (dia >= 1) and (dia <= 31):
      return True
    else:
      return False
  elif mes in [4,6,9,11]:
    if (dia >= 1) and (dia <= 30):
      return True
    else:
      return False
  elif mes==2:
    if Bissexto(ano)==True:
      if (dia >= 1) and (dia <= 29):
        return True
      else:
        return False
    else:
      if (dia >= 1) and (dia <= 28):
        return True
      else:
        return False

if ValidaAno(ano)==True and ValidaMes(mes)==True and ValidaDia(dia,mes)==True:
  print(f'A data {dia}/{mes}/{ano} é válida.')
else:
  print(f'A data {dia}/{mes}/{ano} não é válida.')


Digite uma data no formato dd/mm/aaaa: 29/02/2021
A data 29/2/2021 não é válida.


##Tarefa 19 - Faça um Programa que leia um número inteiro maior que 0 e menor que 1000 e imprima a quantidade de centenas, dezenas e unidades do mesmo.
Observando os termos no plural a colocação do "e", da vírgula entre outros. Exemplo:
*   326 = 3 centenas, 2 dezenas e 6 unidades
*   12 = 1 dezena e 2 unidades

Testar com: 326, 300, 100, 320, 310,305, 301, 101, 311, 111, 25, 20, 10, 21, 11, 1, 7 e 16



In [ ]:
num = input("Digite um número inteiro meior que zero e menor que 1000: ")
listaResultado = [0 for _ in range(3)]
listaNumeros = list(num)
for i in range(1, len(listaNumeros)+1):
  listaResultado[-i] = listaNumeros[-i]

centena = int(listaResultado[-3])
dezena = int(listaResultado[-2])
unidade = int(listaResultado[-1])

if centena > 1:
  pluralCentena = "centenas"
else:
  pluralCentena = "centena"

if dezena > 1:
  pluralDezena = "dezenas"
else:
  pluralDezena = "dezena"

if unidade > 1:
  pluralUnidade = "unidades"
else:
  pluralUnidade = "unidade"

if centena > 0:
  print(f"{centena} {pluralCentena}, {dezena} {pluralDezena} e {unidade} {pluralUnidade}.")
elif dezena > 0:
  print(f"{dezena} {pluralDezena} e {unidade} {pluralUnidade}.")
else:
  print(f"{unidade} {pluralUnidade}.")

Digite um número inteiro meior que zero e menor que 1000: 25
2 dezenas e 5 unidades.


##Tarefa 20 - Faça um Programa para um caixa eletrônico. O programa deverá perguntar ao usuário a valor do saque e depois informar quantas notas de cada valor serão fornecidas. As notas disponíveis serão as de 1, 5, 10, 50 e 100 reais. O valor mínimo é de 10 reais e o máximo de 600 reais. O programa não deve se preocupar com a quantidade de notas existentes na máquina.
Exemplo 1: Para sacar a quantia de 256 reais, o programa fornece duas notas de 100, uma nota de 50, uma nota de 5 e uma nota de 1;
  
Exemplo 2: Para sacar a quantia de 399 reais, o programa fornece três notas de 100, uma nota de 50, quatro notas de 10, uma nota de 5 e quatro notas de 1.

In [ ]:
def Plural(numNotas):
  if numNotas in [1,0]:
    return "nota"
  else:
    return "notas"

valorSacado = int(input("Digite o valor a ser sacado (entre 10 e 600 reais): R$"))

if (valorSacado <10) or (valorSacado >600):
  print("Valor inválido.")
else:
  nota100 = valorSacado//100
  resto100 = valorSacado-(nota100*100)
  nota50 = resto100//50
  resto50 = resto100-(nota50*50)
  nota10 = resto50//10
  resto10 = resto50-(nota10*10)
  nota5 = resto10//5
  resto5 = resto10-(nota5*5)
  nota1 = resto5//1

  numeroExtenso = {0: "nenhuma", 1: "uma", 2: "duas", 3: "três", 4: "quatro", 5: "cinco", 6: "seis"}
  
  print(f"Para sacar a quantia de {valorSacado} reais, o programa fornece: "
  f'\n{numeroExtenso.get(nota100)} {Plural(nota100)} de 100'
  f'\n{numeroExtenso.get(nota50)} {Plural(nota50)} de 50'
  f'\n{numeroExtenso.get(nota10)} {Plural(nota10)} de 10'
  f'\n{numeroExtenso.get(nota5)} {Plural(nota5)} de 5'
  f'\n{numeroExtenso.get(nota1)} {Plural(nota1)} de 1')


Digite o valor a ser sacado (entre 10 e 600 reais): R$598
Para sacar a quantia de 598 reais, o programa fornece: 
cinco notas de 100
uma nota de 50
quatro notas de 10
uma nota de 5
três notas de 1


##Tarefa 21 - Faça um Programa que peça um número inteiro e determine se ele é par ou impar. Dica: utilize o operador módulo (resto da divisão).

In [ ]:
num = int(input("Digite um número inteiro: "))

if (num%2)==0:
  print("É Par")
else:
  print("É Ímpar")

Digite um número inteiro: 53
É Ímpar


##Tarefa 22 - Faça um Programa que peça um número e informe se o número é inteiro ou decimal. Dica: utilize uma função de arredondamento.

In [ ]:
num = float(input("Digite um número: "))

if (num%1) != 0.0:
  print("O número é decimal")
else:
  print("O número é inteiro")

Digite um número: 2.6
O número é decimal


##Tarefa 23 - Faça um Programa que leia 2 números e em seguida pergunte ao usuário qual operação ele deseja realizar. O resultado da operação deve ser acompanhado de uma frase que diga se o número é:
  par ou ímpar;

  positivo ou negativo;
  
  inteiro ou decimal.

In [ ]:
import operator

num1 = float(input("Digite o primeiro número: "))
num2 = float(input("Digite o segundo número: "))
operacao = int(input("Qual operação deseja realizar? 1(+), 2(-), 3(*) e 4(/): "))
op = {1: operator.add, 2: operator.sub, 3: operator.mul, 4: operator.truediv}
op_func = op[operacao]

if operacao not in [1,2,3,4]:
  print("Valor errado para operação.")
else:
  if (operacao==4) and (num2==0.0):
    print("Não é possível dividir por zero.")
  else:
    resultado = op_func(num1, num2)

print(f"O resultado é {resultado:.2f}")

if (resultado%2)==0:
  print("O resultado é Par")
else:
  print("O resultado é Ímpar")

if resultado>=0:
  print("O resultado é positivo")
else:
  print("O resultado é negativo")

if (resultado%1) != 0.0:
  print("O resultado é decimal")
else:
  print("O resultado é inteiro")

Digite o primeiro número: 5
Digite o segundo número: 12
Qual operação deseja realizar? 1(+), 2(-), 3(*) e 4(/): 2
O resultado é -7.00
O resultado é Ímpar
O resultado é negativo
O resultado é inteiro


##Tarefa 24 - Faça um programa que faça 5 perguntas para uma pessoa sobre um crime. As perguntas são:
  "Telefonou para a vítima?"

  "Esteve no local do crime?"
  
  "Mora perto da vítima?"
  
  "Devia para a vítima?"
  
  "Já trabalhou com a vítima?"
 
 O programa deve no final emitir uma classificação sobre a participação da pessoa no crime. Se a pessoa responder positivamente a 2 questões ela deve ser classificada como "Suspeita", entre 3 e 4 como "Cúmplice" e 5 como "Assassino". Caso contrário, ele será classificado como "Inocente".

In [ ]:
positivos = 0
resp =  input("Telefonou para a vítima? (S/N): ")
if resp.upper()=="S":
  positivos += 1
resp =  input("Esteve no local do crime? (S/N): ")
if resp.upper()=="S":
  positivos += 1
resp =  input("Mora perto da vítima? (S/N): ")
if resp.upper()=="S":
  positivos += 1
resp =  input("Devia para a vítima? (S/N): ")
if resp.upper()=="S":
  positivos += 1
resp =  input("Já trabalhou com a vítima? (S/N): ")
if resp.upper()=="S":
  positivos += 1

if positivos==2:
  print("Suspeito")
elif positivos in [3,4]:
  print("Cúmplice")
elif positivos==5:
  print("Assassino")
else:
  print("Inocente")

Telefonou para a vítima? (S/N): n
Esteve no local do crime? (S/N): s
Mora perto da vítima? (S/N): n
Devia para a vítima? (S/N): s
Já trabalhou com a vítima? (S/N): s
Cúmplice


##Tarefa 25 - Um posto está vendendo combustíveis com a seguinte tabela de descontos:

###Álcool:
  *   até 20 litros, desconto de 3% por litro
  *   acima de 20 litros, desconto de 5% por litro

###Gasolina:
*   até 20 litros, desconto de 4% por litro
*   acima de 20 litros, desconto de 6% por litro 

Escreva um algoritmo que leia o número de litros vendidos, o tipo de combustível (codificado da seguinte forma: A-álcool, G-gasolina), calcule e imprima o valor a ser pago pelo cliente sabendo-se que o preço do litro da gasolina é 2,50 reais o preço do litro do álcool é 1,90 reais.

In [ ]:
litros = float(input("Quantos litros? "))
comb = input("Qual combustível? (G-Gasolina, A-Álcool)")

if comb.upper() not in ["A", "G"]:
  print("Combustível inválido.")
else:
  if comb.upper() == "A":
    preco = 1.90
    precobruto = litros * preco
    if litros > 20:
      precoLiq = precobruto * 0.95  
    else:
      precoLiq = precobruto * 0.97
    print(f"O preço a pagar é R${precoLiq:.2f}")
  else:
    preco = 2.50
    precobruto = litros * preco
    if litros > 20:
      precoLiq = precobruto * 0.94  
    else:
      precoLiq = precobruto * 0.96
    
    print(f"O preço a pagar é R${precoLiq:.2f}")



Quantos litros? 56
Qual combustível? (G-Gasolina, A-Álcool)g
O preço a pagar é R$131.60


##Tarefa 26 - Uma fruteira está vendendo frutas com a seguinte tabela de preços:

                    Até 5 Kg                Acima de 5 Kg
    Morango         R$ 2,50 por Kg          R$ 2,20 por Kg
    Maçã            R$ 1,80 por Kg          R$ 1,50 por Kg

Se o cliente comprar mais de 8 Kg em frutas ou o valor total da compra ultrapassar R$ 25,00, receberá ainda um desconto de 10% sobre este total. Escreva um algoritmo para ler a quantidade (em Kg) de morangos e a quantidade (em Kg) de maças adquiridas e escreva o valor a ser pago pelo cliente.

In [ ]:
pesoMaca = float(input("Dê o peso em kg de maçã: "))
pesoMorango = float(input("Dê o peso em kg de morango: "))

if pesoMaca <= 5.0:
  precoMaca = 1.80
else:
  precoMaca = 1.50

if pesoMorango <= 5.0:
  precoMorango = 2.5
else:
  precoMorango = 2.20

pesoTotal = pesoMaca + pesoMorango
preçoTotal = (precoMaca*pesoMaca) + (precoMorango*pesoMorango)

if (pesoTotal > 8.0) or (preçoTotal > 25.0):
  precoTotal = preçoTotal *0.9

print(f"O preço a ser pago pela compra toda é R${preçoTotal:.2f}")

Dê o peso em kg de maçã: 15
Dê o peso em kg de morango: 16
O preço a ser pago pela compra toda é R$57.70


##Tarefa 27 - O Hipermercado Tabajara está com uma promoção de carnes que é imperdível. Confira:

                    Até 5 Kg                Acima de 5 Kg
    File Duplo      R$ 4,90 por Kg          R$ 5,80 por Kg
    Alcatra         R$ 5,90 por Kg          R$ 6,80 por Kg
    Picanha         R$ 6,90 por Kg          R$ 7,80 por Kg

Para atender a todos os clientes, cada cliente poderá levar apenas um dos tipos de carne da promoção, porém não há limites para a quantidade de carne por cliente. Se compra for feita no cartão Tabajara o cliente receberá ainda um desconto de 5% sobre o total da compra. Escreva um programa que peça o tipo e a quantidade de carne comprada pelo usuário e gere um cupom fiscal, contendo as informações da compra: tipo e quantidade de carne, preço total, tipo de pagamento, valor do desconto e valor a pagar.

In [ ]:
tipoCarne = int(input("Qual carne você quer comprar? (1-Filé Duplo, 2-Alcatra e 3-Picanha)"))
pesoCarne = float(input("Quantos quilos quer comprar? "))
cartao = input("Pagará com o nosso cartão? (S/N)")

if (tipoCarne not in [1, 2, 3]) or (cartao.upper() not in ["S", "N"]):
  print("Você inseriu respostas fora dos parâmetros.")
else:
  if(tipoCarne == 1):
    if pesoCarne <= 5.0:
      preco = 4.90
    else:
      preco = 5.80
  elif(tipoCarne == 2):
    if pesoCarne <= 5.0:
      preco = 5.90
    else:
      preco = 6.80
  else:
    if pesoCarne <= 5.0:
      preco = 6.90
    else:
      preco = 7.80   
  
  precoTotal = pesoCarne*preco
  if cartao.upper() == "S":
    precoTotal = precoTotal*0.95

print(f"O preço total da compra é R${precoTotal:.2f}")

Qual carne você quer comprar? (1-Filé Duplo, 2-Alcatra e 3-Picanha)2
Quantos quilos quer comprar? 25
Pagará com o nosso cartão? (S/N)s
O preço total da compra é R$161.50


#**ESTRUTURA DE REPETIÇÃO**

##Tarefa 01 - Faça um programa que peça uma nota, entre zero e dez. Mostre uma mensagem caso o valor seja inválido e continue pedindo até que o usuário informe um valor válido.

In [ ]:
nota = float(input("Digite uma nota de zero a dez: "))

while nota<0 or nota>10:
  print("Você inseriu uma nota errada.")
  nota = float(input("Digite uma nota de zero a dez: "))

Digite uma nota de zero a dez: 11
Você inseriu uma nota errada.
Digite uma nota de zero a dez: 10


##Tarefa 02 - Faça um programa que leia um nome de usuário e a sua senha e não aceite a senha igual ao nome do usuário, mostrando uma mensagem de erro e voltando a pedir as informações.

In [ ]:
login = input("Digite um nome de usuário: ")
senha = input("Digite uma senha diferente do seu nome de usuário: ")

while login == senha:
  print("Você digitou uma senha igual ao seu nome de usuário.")
  login = input("Digite um nome de usuário: ")
  senha = input("Digite uma senha diferente do seu nome de usuário: ")

Digite um nome de usuário: THIAGO
Digite uma senha diferente do seu nome de usuário: THIAGO
Você digitou uma senha igual ao seu nome de usuário.
Digite um nome de usuário: THIAGO
Digite uma senha diferente do seu nome de usuário: 555


##Tarefa 03 - Faça um programa que leia e valide as seguintes informações:
Nome: maior que 3 caracteres;

Idade: entre 0 e 150;

Salário: maior que zero;

Sexo: 'f' ou 'm';

Estado Civil: 's', 'c', 'v', 'd';

In [ ]:
nome = input("Nome (maior que 3 caracteres): ")
while (len(nome) < 4):
  nome = input("Nome inválido, digite novamente: ")

idade = int(input("Idade (entre 0 e 150): "))
while idade <= 0 or idade >= 150:
  idade = int(input("Idade inválida, digite novamente: "))

salario = float(input("Salário (maior que zero): "))
while salario <= 0:
  salario = float(input("Salário inválido, digite novamente: "))

sexo = input("Sexo (M/F): ")
while sexo.upper() != "F" and  sexo.upper() != "M":
  sexo = input("Entrada de sexo inválido, digite novamente: ")

estadoCivil = input("Estado Civil (S, C, V ou D): ")
while estadoCivil.upper() not in ["S", "C", "V", "D"]:
  estadoCivil = input("Entrada de estado civil inválido, digite novamente: ")

Nome (maior que 3 caracteres): TA
Nome inválido, digite novamente: THIAGO
Idade (entre 0 e 150): -5
Idade inválida, digite novamente: 30
Salário (maior que zero): 0
Salário inválido, digite novamente: 7000
Sexo (M/F): T
Entrada de sexo inválido, digite novamente: M
Estado Civil (S, C, V ou D): R
Entrada de estado civil inválido, digite novamente: D


##Tarefa 04 - Supondo que a população de um país A seja da ordem de 80000 habitantes com uma taxa anual de crescimento de 3% e que a população de B seja 200000 habitantes com uma taxa de crescimento de 1.5%. Faça um programa que calcule e escreva o número de anos necessários para que a população do país A ultrapasse ou iguale a população do país B, mantidas as taxas de crescimento.

In [ ]:
popA = 80000
popB = 200000

taxaCrescA = 1.03
taxaCrescB = 1.015
anos = 0

while popA < popB:
  popA *= taxaCrescA
  popB *= taxaCrescB
  anos += 1

print(f'Levou {anos} anos.')
print(f'Hoje o país A possui {popA:.0f} habitantes e o país B {popB:.0f}.')

Levou 63 anos.
Hoje o país A possui 515033 habitantes e o país B 510964.


##Tarefa 05 - Altere o programa anterior permitindo ao usuário informar as populações e as taxas de crescimento iniciais. Valide a entrada e permita repetir a operação.

In [ ]:
popA = int(input("Digite o número de habitantes do país A: "))
while popA <1:
  print("Existe pelo menos um habitante.")
  popA = int(input("Digite o número de habitantes do país A: "))

popB = int(input("Digite o número de habitantes do país B: "))
while popB <1:
  print("Existe pelo menos um habitante.")
  popB = int(input("Digite o número de habitantes do país B: "))

taxaCrescA = float(input("Digite a taca de crescimento do país A em porcentagem: "))
taxaCrescA = (taxaCrescA/100)+1
taxaCrescB = float(input("Digite a taca de crescimento do país B em porcentagem: "))
taxaCrescB = (taxaCrescB/100)+1
anos = 0

while popA < popB:
  popA *= taxaCrescA
  popB *= taxaCrescB
  anos += 1

print(f'Levou {anos} anos.')
print(f'Hoje o país A possui {popA:.0f} habitantes e o país B {popB:.0f}.')

Digite o número de habitantes do país A: 45000
Digite o número de habitantes do país B: 150000
Digite a taca de crescimento do país A em porcentagem: 8
Digite a taca de crescimento do país B em porcentagem: 3
Levou 26 anos.
Hoje o país A possui 332836 habitantes e o país B 323489.
